test/Microsoft.Health.Events.UnitTest/EventBatchingTests.cs

```diff
@@ -0,0 +1,194 @@
+using Microsoft.Health.Events.EventCheckpointing;
+using Microsoft.Health.Events.EventConsumers.Service;
+using Microsoft.Health.Events.Model;
+using Microsoft.Health.Logging.Telemetry;
+using NSubstitute;
+using System;
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using System.Linq;
+using Xunit;
+
+namespace Microsoft.Health.Events.UnitTest
+{
+    public class EventBatchingTests
+    {
+        [Fact]
+        public async void GivenBatchServiceCreated_WhenFirstEventReceived_ThenCreateQueueAndWindow_Test()
+        {
+            var eventConsumerService = Substitute.For<IEventConsumerService>();
```

Comment:
Just a quick note, I see these objects getting created at the start of each test. You can actually do this in the constructor of the test class and set private fields that each test can use. The constructor is called at the start of each test so there is no concern of shared context between tests. The behavior is essentially the same, it's just less lines of code.

```c#
private IEventConsumerService _eventConsumerService;
private ICheckpointClient _checkpointClient;
private ITelemetryLogger _logger;
private EventBatchingOptions _options;

public EventBatchingTests()
{
     _eventConsumerService = Substitute.For<IEventConsumerService>();
     _checkpointClient = Substitute.For<ICheckpointClient>();
     _logger = Substitute.For<ITelemetryLogger>();
     _options = new EventBatchingOptions();
}
```

Reply:
Thanks for the suggestion. Updated.